datalad/distribution/create.py

```diff
@@ -10,78 +10,205 @@
-            without any annex""",
+            doc="""if set, a plain Git repository will be created without any
+            annex""",
             action='store_false'),
         annex_version=Parameter(
             args=("--annex-version",),
-            doc="""select particular annex repository version.  The list of
-            supported versions depends on the available git-annex version""",
+            doc="""select a particular annex repository version. The
+            list of supported versions depends on the available git-annex
+            version. This should be left untouched, unless you know what
+            you are doing""",
             constraints=EnsureDType(int) | EnsureNone()),
         annex_backend=Parameter(
             args=("--annex-backend",),
+            constraints=EnsureStr() | EnsureNone(),
             # not listing choices here on purpose to avoid future bugs
             doc="""set default hashing backend used by the new dataset.
             For a list of supported backends see the git-annex
-            documentation""",
+            documentation. The default is optimized for maximum compatibility
+            of datasets across platforms (especially those with limited
+            path lengths)""",
             nargs=1),
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_init_opts=annex_init_opts,
     )
 
     @staticmethod
-    @datasetmethod(name='create', dataset_argname='loc')
+    @datasetmethod(name='create', dataset_argname='path')
     def __call__(
-            loc=None, description=None, no_annex=False, annex_version=None,
-            annex_backend='MD5E'):
-        if description and no_annex:
-            raise ValueError("Incompatible arguments: cannot specify description for "
-                             "annex repo and declaring no annex repo.")
-        if loc is None:
-            loc = os.curdir
-        elif isinstance(loc, Dataset):
-            loc = loc.path
-        if no_annex:
-            lgr.info("Creating a new git repo at %s", loc)
-            vcs = GitRepo(loc, url=None, create=True)
+            path=None,
+            force=False,
+            description=None,
+            add_to_super=False,
+            name=None,
+            no_annex=False,
+            annex_version=None,
+            annex_backend='MD5E',
+            git_opts=None,
+            annex_opts=None,
+            annex_init_opts=None):
+
+        if path:
+            if isinstance(path, Dataset):
+                ds = path
+            else:
+                # this means we are not bound to a Dataset instance.
+                # Therefore there is no difference between a relative path and
+                # an "explicit" path. Both are based on CWD, since there is no
+                # repo or dataset it is passed to. Hence `path` doesn't need to
+                # be resolved.
+                ds = Dataset(path)
```

Comment:
ah -- I see now a reason -- path is optional and name is (imho incorrectly, especially since "Used only together with `add_to_super`.""",) mandatory!:
```shell
$> datalad create
usage: datalad create [--version] [-h]
                      [-l {critical,error,warning,info,debug,1,2,3,4,5,6,7,8,9}]
                      [-p {condor}] [-f] [-D DESCRIPTION] [--add-to-super]
                      [--no-annex] [--annex-version ANNEX_VERSION]
                      [--annex-backend ANNEX_BACKEND] [--git-opts STRING]
                      [--annex-opts STRING] [--annex-init-opts STRING]
                      [PATH] NAME
$> datalad create /tmp/ds2 blah
2016-07-07 13:46:31,873 [INFO   ] Creating a new annex repo at /tmp/ds2 (create.py:194)
Created dataset at /tmp/ds2.
```

needs a fix and a test for such a rudimentary usecase